src/garage/envs/grid_world_env.py

```diff
@@ -30,114 +31,185 @@
 }   # yapf: disable
 
 
-class GridWorldEnv(gym.Env):
-    """
+class GridWorldEnv(Environment):
+    """A simply 2D grid environment.
+
     | 'S' : starting point
     | 'F' or '.': free space
     | 'W' or 'x': wall
     | 'H' or 'o': hole (terminates episode)
     | 'G' : goal
     """
 
-    def __init__(self, desc='4x4'):
+    def __init__(self, desc='4x4', max_episode_length=None):
+        """Initialize the environment.
+
+        Args:
+            desc (str): grid configuration key.
+            max_episode_length (int): The maximum steps allowed for an episode.
+        """
         if isinstance(desc, str):
             desc = MAPS[desc]
         desc = np.array(list(map(list, desc)))
         desc[desc == '.'] = 'F'
         desc[desc == 'o'] = 'H'
         desc[desc == 'x'] = 'W'
-        self.desc = desc
-        self.n_row, self.n_col = desc.shape
+        self._desc = desc
+        self._n_row, self._n_col = desc.shape
```

Comment:
```suggestion
        self._n_row = desc.shape
        self._n_col = desc.shape
```

Reply:
Not sure I get this. why would we assign a tuple to `_n_row` and `_n_col`?

Reply:
oops -- sorry i messed up